swarm-test/src/lib.rs

```diff
@@ -21,14 +21,10 @@
 use async_trait::async_trait;
 use futures::future::{BoxFuture, Either};
 use futures::{FutureExt, StreamExt};
-use libp2p_core::{
-    multiaddr::Protocol, transport::MemoryTransport, upgrade::Version, Multiaddr, Transport,
-};
-use libp2p_identity::{Keypair, PeerId};
-use libp2p_plaintext as plaintext;
+use libp2p_core::{multiaddr::Protocol, Multiaddr};
+use libp2p_identity::PeerId;
 use libp2p_swarm::dial_opts::PeerCondition;
-use libp2p_swarm::{self as swarm, dial_opts::DialOpts, NetworkBehaviour, Swarm, SwarmEvent};
-use libp2p_yamux as yamux;
```

Comment:
can we maintain these imports and renames?

Reply:
Same, these imports are only used in `fn new_ephemeral` and `fn new_ephemeral_tokio`. Disabling both would cause clippy warnings of used imports